horovod/common/operations.cc

```diff
@@ -1630,9 +1483,12 @@ bool RunLoopOnce(HorovodGlobalState& state, MPIContext& ctx,
     std::string encoded_response;
     ResponseList::SerializeToString(response_list, encoded_response);
     int encoded_response_length = (int)encoded_response.length() + 1;
-    MPI_Bcast(&encoded_response_length, 1, MPI_INT, RANK_ZERO, ctx.mpi_comm);
-    MPI_Bcast((void*)encoded_response.c_str(), encoded_response_length,
-              MPI_BYTE, RANK_ZERO, ctx.mpi_comm);
+    state.controller->Bcast(&encoded_response_length, 1, HOROVOD_INT32,
```

Comment:
Similar to above, here we're sending the final list of tensors we want all ranks to reduce.  So I'd put this behind a method called `SendFinalTensors()`.

Reply:
Done.